test/bdd/fixtures/agent-rest/docker-compose.yml

```diff
@@ -252,6 +252,58 @@ services:
     networks:
       - bdd_net
 
+  couchdb.example.com:
+    container_name: couchdb.example.com
+    image: ${COUCHDB_IMAGE}:${COUCHDB_IMAGE_TAG}
+    ports:
+      - ${COUCHDB_PORT}:${COUCHDB_PORT}
+    environment:
+      - COUCHDB_USER=${COUCHDB_USERNAME}
+      - COUCHDB_PASSWORD=${COUCHDB_PASSWORD}
+    volumes:
+      - ./couchdb-config/10-single-node.ini:/opt/couchdb/etc/local.d/config.ini
+    networks:
+      - bdd_net
+
+  kms.example.com:
+    container_name: kms.example.com
+    image: ${KMS_REST_IMAGE}:${KMS_REST_TAG}
+    environment:
+      - KMS_HOST_URL=0.0.0.0:8076
+      - KMS_BASE_URL=https://kms.example.com:8076
+      - KMS_TLS_CACERTS=/etc/tls/ec-cacert.pem
+      - KMS_TLS_SERVE_CERT=/etc/tls/ec-pubCert.pem
+      - KMS_TLS_SERVE_KEY=/etc/tls/ec-key.pem
+      - KMS_DATABASE_TYPE=couchdb
+      - KMS_DATABASE_URL=${COUCHDB_USERNAME}:${COUCHDB_PASSWORD}@couchdb.example.com:${COUCHDB_PORT}
+      - KMS_DATABASE_PREFIX=keystore
+      - KMS_SECRET_LOCK_KEY_PATH=/etc/tls/secret-lock.key
+      - KMS_PRIMARY_KEY_DATABASE_TYPE=couchdb
+      - KMS_PRIMARY_KEY_DATABASE_URL=${COUCHDB_USERNAME}:${COUCHDB_PASSWORD}@couchdb.example.com:${COUCHDB_PORT}
+      - KMS_PRIMARY_KEY_DATABASE_PREFIX=kmspk
+      - KMS_LOCAL_KMS_DATABASE_TYPE=couchdb
+      - KMS_LOCAL_KMS_DATABASE_URL=${COUCHDB_USERNAME}:${COUCHDB_PASSWORD}@couchdb.example.com:${COUCHDB_PORT}
+      - KMS_LOCAL_KMS_DATABASE_PREFIX=kmslocal
+      - KMS_KEY_MANAGER_STORAGE_TYPE=couchdb
+      - KMS_KEY_MANAGER_STORAGE_URL=${COUCHDB_USERNAME}:${COUCHDB_PASSWORD}@couchdb.example.com:${COUCHDB_PORT}
+      - KMS_KEY_MANAGER_STORAGE_PREFIX=kmskm
+      #- KMS_KEY_MANAGER_STORAGE_TYPE=edv
+      #- KMS_KEY_MANAGER_STORAGE_URL=https://edv.example.com:8081
+      - KMS_CACHE_EXPIRATION=10m
+      - KMS_ZCAP_ENABLE=false
+      - KMS_LOG_LEVEL=debug
+    ports:
+      - 8076:8076
+    entrypoint: ""
+    # TODO add retry for couchdb
```

Comment:
Are these BDD tests flaky? We're already having enough of that flakiness in our CI.

Reply:
this was copied from hub-kms .. 
the key server depends the added couchdb instance, I did add `depends_on`, but in case the container doesn't start, we might need to do retries